cpp/include/cudf/fixed_point/fixed_point.hpp

```diff
@@ -537,21 +537,23 @@ class fixed_point {
   explicit operator std::string() const
   {
     if (_scale < 0) {
-      int const n = std::pow(10, -_scale);
-      int const f = _value % n;
+      auto const av = std::abs(_value);
+      int const n   = std::pow(10, -_scale);
+      int const f   = av % n;
       auto const num_zeros =
         std::max(0, (-_scale - static_cast<int32_t>(std::to_string(f).size())));
       auto const zeros = std::string(num_zeros, '0');
-      return std::to_string(_value / n) + std::string(".") + zeros +
-             std::to_string(std::abs(_value) % n);
+      auto const sign  = _value < 0 ? std::string("-") : std::string();
+      return sign + std::to_string(av / n) + std::string(".") + zeros + std::to_string(av % n);
```

Comment:
Does `std::to_string(_value / n)` not properly convert the sign to a `-`?

Reply:
It does. But if `_value < n` then `(_value / n )` is 0 and the sign is lost.
For example, `-0.012` can be fixed-point stored as `_value = -12` with `scale=-3`. 
This gives `n=1000` and so `(_value / n)` will lose the sign.

Reply:
Ah, nice catch.